sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -487,22 +481,41 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
                 return getPublicClientApplication(false)
                     .acquireTokenSilently(parametersBuilder.build());
             } catch (MalformedURLException e) {
-                throw logger.logExceptionAsError(Exceptions.propagate(e));
+                return getFailedCompletableFuture(logger.logExceptionAsError(Exceptions.propagate(e)));
```

Comment:
@alzimmermsft suggested this usage pattern to me.

Reply:
Actually I know a reason - if `e` is already a subclass of `RuntimeException`, wrapping it in another `RuntimeException` is bad. `Exceptions.propagate(e)` will check if `e` is a RuntimeException first and only wrap if not.

Reply:
Okay,
But we already know, MalformedURLException is not a RuntimeException here.
So, that check is redundant.

Reply:
That's correct. But at this point it's really a personal preference thing. I changed it to `new RuntimeException()` for the if check it saves. But one could argue in the future if more exceptions are thrown or this code path is refactored to merge with other paths that could throw RuntimeExceptions this could potentially be a point of failure. The point is, it's not critical.